torch_np/_ndarray.py

```diff
@@ -343,67 +399,80 @@ def __setitem__(self, index, value):
         return self._tensor.__setitem__(index, value)
 
 
-def asarray(a, dtype=None, order=None, *, like=None):
-    _util.subok_not_ok(like)
-    if order is not None:
+# This is the ideally the only place which talks to ndarray directly.
+# The rest goes through asarray (preferred) or array.
+
+def array(object, dtype=None, *, copy=True, order='K', subok=False, ndmin=0,
+          like=None):
+    _util.subok_not_ok(like, subok)
+    if order != 'K':
```

Comment:
refactor into a `_util` function?

Reply:
Definitely. In a separate PR a bit later down the line, I'd say. Across the whole codebase